### PR TITLE
feat(cb2-2522): adds big unit test

### PR DESCRIPTION
**Pull request by andrewf-bjss**
Adds a "big unit test" test to check the entirety of the lambda's functions and internal flow by mocking all calls to external dependencies:
- AWS Secrets Manager
- MSAL (Azure AD Authentication Token request)
- DynamicsCE OData - data source: 4 records returned
- AWS EventBridge - data destination: 3 converted events received

[CB2-2522)](https://jira.dvsacloud.uk/browse/CB2-2522)

## Checklist
- [x] Code has been tested manually
- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -20,8 +20,8 @@
     "test:unit": "cross-env NODE_ENV=test API_VERSION=${npm_package_version} jest --runInBand",
     "test:unit:watch": "npm run test:unit -- --watch",
     "test:unit:coverage": "cross-env npm run test:unit -- --coverage",
-    "test-i": "echo 'to be added: launch server' && npm run test:integration",
-    "test:integration": "cross-env NODE_ENV=test echo 'integration tests to be added'",
+    "test-i": "npm run test:integration",
```
Comment (premghinde) [inline package.json:23]: Can you make this follow the naming convention?

**Review by premghinde — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (andrewf-bjss) [reply package.json:23]: Sorry @premghinde - could you elaborate please? I'm not sure how you want them changed. These scripts are from [the template repo](https://github.com/dvsa/cvs-svc-template)

**Review by Ian-Hodges — COMMENTED** [review]
I might be missing something, but at what point does this test any integration with external systems? It all seems to be mocked.

File: package.json (hunk 1 above)
Comment (premghinde) [reply package.json:23]: All other commands are separated with `:`'s creating a de facto naming standard. The hypenated one is the odd one out. I would suggest a fix in the CVS template also.

Comment (Ian-Hodges) [reply package.json:23]: Potential complication: If we did change it, we would need to update the pipeline too. We would then have to manage two versions until all repos have been converted.

Comment (premghinde) [reply package.json:23]: Please add a CI ticket for this

**Comment by andrewf-bjss** [comment]
Following team discussions and considering time constraints and release pipelines currently running `npm run test-i`, the decision has been made to accept this PR as-is, which is essentially a "big unit test", rather than our agreed definition of an integration test being the external dependency responses are mocked - rather than the external dependencies themselves so the calls actually leave the lambda code.

**Review by Ian-Hodges — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (andrewf-bjss) [reply package.json:23]: @premghinde Task raised to create a CI ticket for the change requested: [CB2-2595](https://jira.dvsacloud.uk/browse/CB2-2595)

**Pull request merged**